tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+    client.run('yum -y install subscription-manager')
+    sub_man = client.package('subscription-manager')
+    assert sub_man.is_installed
+
+    hammer = smoker.package('tfm-rubygem-hammer_cli_katello')
+    assert hammer.is_installed
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_username_and_password(client, entities, server_hostname):
+    client.run("yum erase -y 'katello-ca-consumer-*'")
+
+    client.run_expect([0], "rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" % server_hostname)
+    client.run_expect([0], "subscription-manager register --force --org=%s --username=admin --password=changeme --env=Library" % entities['organization_label'])
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_activation_key(client, entities):
+    client.run_expect([0], "subscription-manager register --force --org=%s --activationkey=\"%s\"" % (entities['organization_label'], entities['activation_key']))
+
+    consumed = client.check_output("subscription-manager list --consumed")
+    assert entities['product'] in consumed
+
+
+@pytest.mark.katello_agent
+def test_host_is_registered(client, smoker):
+    hostname = client.check_output('hostname -s')
+    smoker.run_expect([0], "hammer host info --name %s" % hostname)
+
+
+@pytest.mark.katello_agent
+def test_enable_content_view_repo(client, entities):
+    enable = client.check_output("subscription-manager repos --enable=%s_%s_%s" % (entities['organization_label'], entities['product_label'], entities['yum_repository_label']))
+    assert "is enabled for this system" in enable
+
+
+def test_install_katello_host_tools(client, foreman_version):
+    release = client.system_info.release.split('.')[0]
+    client.run_expect([0], "yum -y install http://yum.theforeman.org/client/%s/el%s/x86_64/foreman-client-release.rpm" % (foreman_version, release))
+    client.run_expect([0], 'yum -y install katello-host-tools')
+    katello_host_tools = client.package('katello-host-tools')
+    assert katello_host_tools.is_installed
+
+
+def test_install_package_local(client):
+    client.run('yum -y install walrus-0.71')
+    walrus = client.package('walrus')
+    assert walrus.is_installed
+    assert walrus.version == '0.71'
+
+
+def test_available_errata(client, smoker, entities):
+    time.sleep(10)
+    hostname = client.check_output('hostname -s')
+    errata = smoker.check_output("hammer host errata list --host %s" % hostname)
+    assert entities['errata_id'] in errata
+
+
+@pytest.mark.katello_agent
+def test_install_katello_agent(client):
+    release = client.system_info.release.split('.')[0]
+
+    if release != 6 or release != 7:
+        pytest.skip("katello-agent not available on EL8")
+
+    client.run_expect([0], 'yum -y install katello-agent')
+    katello_agent = client.package('katello-agent')
+    assert katello_agent.is_installed
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_package_install(client, smoker):
```

Comment:
I think this test depends on `katello-agent` being installed. I don't think you can really depend on another test and if we use xdist to run tests concurrently this can easily break. I'd suggest to do the installation as part of the fixture so you know for sure it's installed.